### DISCUSSION TITLE
Using Falcon as a reverse proxy (http or https)

### DISCUSSION THREAD

**Discussion by henri** [socketry/falcon#262] [Q&A] [answered]
I am curious to know if falcon or perhaps falcon combined with other socketry components would be suitable for use a quick to setup reverse proxy?

For example, if I was to use NGNIX with the following configuration : 

server {
    listen 3001;

    server_name _;

    location / {
        proxy_pass http://localhost:3000;

        proxy_set_header X-Custom-Header "MyValue"; # custom header setup

        proxy_set_header X-Real-IP $remote_addr;  # pass client IP
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;  # original client IP in the header
    }
}

How would I approach this with Falcon or would I need to look at building or using some other tools to get this working?

Thanks and hopefully this is something which Falcon is already capable of handling? If this functionality is not yet ready I would be happy to help out to get it going even if that is a separate project, wrapper or set of scripts?

My end goal is to be able to easily setup a system to put up / tear down a set of reverse proxies.

**Comment by ioquatix** [comment, marked as answer]
I have added an example here showing how to implement a proxy: https://github.com/socketry/falcon/blob/main/examples/proxy/falcon.rb
